duckduckgo/iOS
# Performance improvements

Task/Issue URL: https://app.asana.com/0/856498667320406/1127139659426257
Tech Design URL: https://app.asana.com/0/856498667320406/1128499240567703

**Description**:
Created Cache to improve overall Tracker Blocking performance across the app.

**Steps to test this PR**:

Fresh install.
1. Install DDG app.
2. First run should download content blocker lists.
3. Content blocking&privacy grade should work.

Fetch missing lists
1. Install DDG app.
2. Open it while blocking network connections.
4. Navigate to some page - privacy grade should be 'unknown'.
5. Open privacy dashboard - there should be info about missing data.
6. Try fetching data manually - it should reload current page and present privacy grade.

Performance check
1. Open page with many trackers, that has a feed-like content (e.g. cnn.com, wykop.pl).
2. Try scrolling while contents are being loaded.

Privacy screen
- Omni bar should show privacy grade.
- All detail screens should work as expected.

Whitelist
- Whitelist screen should work as expected.
- Whitelisting functionality should work.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
